components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -44,29 +42,45 @@ internal const val LAST_VERIFIED = "last_verified_push_subscription"
  * Defaults to false so that observers are always notified.
  */
 class FxaPushSupportFeature(
-    context: Context,
+    private val context: Context,
     accountManager: FxaAccountManager,
     pushFeature: AutoPushFeature,
     owner: LifecycleOwner = ProcessLifecycleOwner.get(),
     autoPause: Boolean = false
 ) {
+
+    /**
+     * A unique scope for the FxA push subscription.
+     */
+    private val pushScope: String by lazy {
+        // Generate a unique scope if one doesn't exist.
+        val scope = UUID.randomUUID().toString().replace("-", "")
```

Comment:
> Even though this is in a lazy, you should still look it up first in the prefs and avoid generating a random UUID and doing string manipulations on it when you don't need to. 

For `SharedPreferences.getString` I need to provide a default string even if I do the pref check first (and for push that can't be null), so generating a UUID is pretty cheap since it's only done once per-startup.